gossip_tracer.go

```diff
@@ -29,7 +29,7 @@ type gossipTracer struct {
 
 func newGossipTracer() *gossipTracer {
 	return &gossipTracer{
-		msgID:        DefaultMsgIdFn,
+		idGen:        newMsgIdGenerator(),
```

Comment:
Can be removed, as anyway set on Start. Kept matching the existing pattern.

Reply:
Same with other tracers

Reply:
yeah, let's drop this.

Reply:
actually wait, tests might need it so let's keep it.